models/cycle_gan_semantic_model.py

```diff
@@ -225,13 +247,22 @@ def backward_G(self):
         #print('fake_B=',self.pred_fake_B)
         #print('input_A_label=',self.input_A_label)
         #print(self.pred_fake_B.shape,self.input_A_label.shape)
-        self.loss_sem_AB = self.criterionCLS(self.pred_fake_B, self.input_A_label)
+        if not self.opt.regression:
+            self.loss_sem_AB = self.criterionCLS(self.pred_fake_B, self.input_A_label)
+        else:
+            self.loss_sem_AB = self.criterionCLS(self.pred_fake_B.squeeze(1), self.input_A_label)
+                
         #self.loss_sem_AB = self.criterionCLS(self.pred_fake_B, self.gt_pred_A)
         # semantic loss BA
         if hasattr(self,'input_B_label'):
-            self.loss_sem_BA = self.criterionCLS(self.pred_fake_A, self.input_B_label)
-        else:
+            if not self.opt.regression:
+                self.loss_sem_BA = self.criterionCLS(self.pred_fake_A, self.input_B_label)
+            else:
+                self.loss_sem_BA = self.criterionCLS(self.pred_fake_A.squeeze(1), self.input_B_label)
+        elif not self.opt.regression:
```

Comment:
As there are many option and branching everywhere in the code, it's better to be as structured as possible.
Here I would do
```python
if hasattr(self,'input_B_label'):
    if self.opt.regression:
        ...
else:
    if self.opt.regression:
        ...
```
It will make the code more readable in the future